src/main/java/com/wavefront/sdk/jaxrs/client/WavefrontJaxrsClientFilter.java

```diff
@@ -0,0 +1,84 @@
+package com.wavefront.sdk.jaxrs.client;
+
+import com.wavefront.sdk.common.WavefrontSender;
+import com.wavefront.sdk.common.application.ApplicationTags;
+import com.wavefront.sdk.common.application.HeartbeaterService;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+
+import static com.wavefront.sdk.jaxrs.Constants.JAXRS_CLIENT_COMPONENT;
+import static com.wavefront.sdk.jaxrs.Constants.PROPERTY_NAME;
+import static com.wavefront.sdk.jaxrs.Constants.CHILD_OF;
+
+/**
+ * A filter to generate Wavefront client side span for JAX-RS based API requests/responses.
+ *
+ * @author Hao Song (songhao@vmware.com).
+ */
+public class WavefrontJaxrsClientFilter implements ClientRequestFilter, ClientResponseFilter {
+
+  @Nullable
+  private final Tracer tracer;
+  private final List<ClientSpanDecorator> spanDecorators;
+
+  public WavefrontJaxrsClientFilter(WavefrontSender sender, ApplicationTags applicationTags,
+                                    String source, @Nullable Tracer tracer) {
+    this.tracer = tracer;
+    this.spanDecorators = Arrays.asList(ClientSpanDecorator.STANDARD_TAGS,
+        ClientSpanDecorator.WF_PATH_OPERATION_NAME);
+    HeartbeaterService heartbeaterService = new HeartbeaterService(sender, applicationTags,
+        JAXRS_CLIENT_COMPONENT, source);
+    heartbeaterService.run();
```

Comment:
You dont need to invoke this as it is taken care of in 
https://github.com/wavefrontHQ/wavefront-sdk-java/blob/master/src/main/java/com/wavefront/sdk/common/application/HeartbeaterService.java#L50